tests/unit_tests/sql_lab_test.py

```diff
@@ -125,6 +130,90 @@ def test_execute_sql_statement_with_rls(
     SupersetResultSet.assert_called_with([(42,)], cursor.description, db_engine_spec)
 
 
+def test_execute_sql_statement_exceeds_payload_limit_log_check(mocker: MockerFixture, caplog) -> None:
+    """
+    Test for `execute_sql_statements` when the result payload size exceeds the limit,
+    and check if the correct log message is captured without raising the exception.
+    """
```

Comment:
What is the purpose of this test? Shouldn't we always raise an exception if the payload size exceeds the threshold?

Reply:
Updated the test to raise exception and assert on that.